### PR TITLE
Fix full-model benchmark execution contracts

### PR DESCRIPTION
**Pull request by TidalPaladin**
## Motivation

The full-model benchmark measured the same forward-plus-backward workload for both training pass modes even though it reported different operation counts. The command also treated a documented scalar resolution as one-dimensional and returned exit status 0 after every requested case failed. Those behaviors make latency data and automation results unreliable.

Addresses #102.

## Solution

Define the timer boundary for each pass mode, expand scalar inputs to square 2D resolutions, and return explicit CLI status codes. Backward-only now builds its graph before the synchronized timer; `forward_backward` measures graph construction, loss construction, and autograd together. Runtime diagnostics use stderr while the benchmark report and successful case data remain on stdout.

## Changes

- Measure backward-only and combined training workloads with distinct timer boundaries.
- Keep the existing approximate operation multipliers aligned with the measured workloads.
- Reset CUDA peak-memory statistics after backward-only graph construction.
- Interpret `--resolutions 224` as `(224, 224)`.
- Validate device and configuration inputs with concise runtime errors.
- Return exit code 2 when any requested case or plot operation fails, including empty runs.
- Save and report successful cases when another requested case fails.
- Document the three pass-mode contracts.

## Test plan

- [x] `make test-benchmark` (32 passed)
- [x] `make quality`
- [x] `make types`
- [x] `make test-ci` (412 passed, 258 skipped, 6 deselected; 94% production coverage)
- [x] Run `vit-benchmark` with `--device invalid-device` and confirm exit status 2.

## Test suite changes

Added ten parametrized or standalone cases across seven test functions. They cover timer event ordering, operation-count multipliers, scalar resolution expansion, invalid-device handling, successful report output, and all-cases-failed stream and exit behavior. The existing YAML fixture text moved to a shared constant; its assertions and coverage intent did not change.

## Risks

Backward-only measurement still performs an untimed forward pass for each sample because autograd needs a fresh graph. The documented GFLOP multipliers remain estimates rather than profiler-derived counts. No benchmark default or result schema changed.

Generated with Codex.

### CODE REVIEW

**Comment by TidalPaladin** [comment]
@codex review

File: benchmark/benchmark.py [hunk 1]
```diff
@@ -257,17 +257,7 @@ def benchmark_memory(
...
-            # Compute loss
-            if isinstance(output, torch.Tensor):
-                loss = output.mean()
-            else:
-                loss = output.dense_features.mean()
-
-            if pass_mode in ["backward", "forward_backward"]:
-                loss.backward()
-
+            _benchmark_loss(model(input_tensor)).backward()
```
[the remark this answers is not in the document]
Comment (TidalPaladin) [reply benchmark/benchmark.py:270]: Fixed in `725f764`. Backward-only memory measurement now constructs the graph before resetting CUDA peak statistics, then measures the backward phase. The regression asserts the forward, reset, backward, and peak-read ordering without requiring a CUDA runner.
